### PR TITLE
Add integration test to file_system_buffer filter

### PR DESCRIPTION
**Pull request by ravenblackx**
Commit Message: Add integration test to file_system_buffer filter
Additional Description:
Risk Level: None, test-only
Testing: Yes
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: test/extensions/filters/http/file_system_buffer/BUILD [hunk 1]
```diff
@@ -34,3 +34,16 @@ envoy_cc_test(
         "//test/mocks/upstream:upstream_mocks",
     ],
 )
+
+envoy_cc_test(
```
Comment (mattklein123) [inline test/extensions/filters/http/file_system_buffer/BUILD:38]: Sorry I should have caught this before, but for concrete filters, these should be envoy_extension_cc_test and the filter code itself should be extension targets. If there is some reason you are doing this switch in a future PR can you add TODOs in various places?

File: test/extensions/filters/http/file_system_buffer/filter_integration_test.cc [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+          request:
+            memory_buffer_bytes_limit: 64
+            storage_buffer_bytes_limit: 64
+            behavior:
+              fully_buffer: {}
+          storage_buffer_path: )EOF",
+      testTmpDir());
+}
+
+// Integration test //
```
Comment (mattklein123) [inline test/extensions/filters/http/file_system_buffer/filter_integration_test.cc:58]: nit: delete

File: test/extensions/filters/http/file_system_buffer/filter_integration_test.cc [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+using FileSystemBufferIntegrationTest = HttpProtocolIntegrationTest;
```
Comment (mattklein123) [inline test/extensions/filters/http/file_system_buffer/filter_integration_test.cc:60]: Please run these new tests with runs_per_test using the deflake instructions to check for flakes.https://github.com/envoyproxy/envoy/tree/main/test/integration#reproducing-test-flakes

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Awesome thanks for adding the test coverage.

/wait

File: test/extensions/filters/http/file_system_buffer/BUILD (hunk 1 above)
Comment (ravenblackx) [reply test/extensions/filters/http/file_system_buffer/BUILD:39]: Added TODO (it is indeed done in the next PR - the extension doesn't truly have a name until then. Maybe that would be fine, I don't really know what the purpose of the extension name tag is in extension_cc_tests.)

File: source/extensions/filters/http/file_system_buffer/fragment.h [hunk 4]
```diff
@@ -102,6 +102,11 @@ class Fragment {
 private:
   size_t size_;
   absl::variant<MemoryFragment, StorageFragment, ReadingFragment, WritingFragment> data_;
+
+  // Disabling copy and move constructors for safety because long-lived fragment pointers
+  // are used.
+  Fragment(const Fragment&) = delete;
+  Fragment(Fragment&&) = delete;
```
Comment (mattklein123) [inline source/extensions/filters/http/file_system_buffer/fragment.h:109]: You an inherit from NonCopyable (search code) if you want. Feel free to do this in a follow up if you like.

**Review by mattklein123 — COMMENTED** [review]
Thanks LGTM. Can you check tidy?

/wait

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**